phylo/src/alignment/sequences.rs

```diff
@@ -0,0 +1,90 @@
+use crate::alphabets::{detect_alphabet, Alphabet, GAP};
+
+use bio::io::fasta::Record;
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct Sequences {
+    pub(crate) s: Vec<Record>,
+    pub(crate) aligned: bool,
+    pub(crate) msa_len: usize,
+    pub(crate) alphabet: Alphabet,
+}
+
+impl Sequences {
+    /// Creates a new Sequences object from a vector of bio::io::fasta::Record.
+    /// The Sequences object is considered aligned if all sequences have the same length.
+    pub fn new(s: Vec<Record>) -> Sequences {
+        let len = if s.is_empty() { 0 } else { s[0].seq().len() };
+        let alphabet = detect_alphabet(&s);
+        if s.iter().filter(|rec| rec.seq().len() != len).count() == 0 {
+            Sequences {
+                s,
+                aligned: true,
+                msa_len: len,
+                alphabet,
+            }
+        } else {
+            Sequences {
+                s,
+                aligned: false,
+                msa_len: 0,
+                alphabet,
+            }
+        }
+    }
+
+    pub fn iter(&self) -> impl Iterator<Item = &Record> {
+        self.s.iter()
+    }
+
+    pub fn len(&self) -> usize {
+        self.s.len()
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.s.is_empty()
+    }
+
+    pub fn get(&self, idx: usize) -> &Record {
```

Comment:
At least according to ChatGPT, Rust actually prefers getters to be called by the name of the field itself, so I should rather use record() or record_mut(). When I look through the codebase it doesn't seem like there's a lot of ref() and ref_mut() going on. In this case, I prefer
` pub fn record(&self, idx: usize) -> &Record {}`
What do you think?

Reply:
changed to add a non-panicking version of `record_by_id` just in case:

```
    pub fn record(&self, idx: usize) -> &Record {
        &self.s[idx]
    }

    pub fn record_mut(&mut self, idx: usize) -> &mut Record {
        &mut self.s[idx]
    }

    pub fn record_by_id(&self, id: &str) -> &Record {
        self.s
            .iter()
            .find(|r| r.id() == id)
            .unwrap_or_else(|| panic!("Sequence with id {} not found", id))
    }

    pub fn try_record_by_id(&self, id: &str) -> Result<&Record> {
        let rec = self.s.iter().find(|r| r.id() == id);
        match rec {
            Some(r) => Ok(r),
            None => bail!("Sequence with id {} not found", id),
        }
    }
```